ArduPlane/fence.cpp

```diff
@@ -80,6 +81,13 @@ void Plane::fence_check()
         switch (fence_act) {
         case AC_FENCE_ACTION_REPORT_ONLY:
             break;
+#if MODE_AUTOLAND_ENABLED
+        case AC_FENCE_ACTION_AUTOLAND_OR_RTL:
```

Comment:
I think you need the same check as on line 95. Probably easier to fall through to that case and then add the if on 94.

Reply:
You also need to protect against RLT vs AUTOLAND. It should not switch out of one to go into the other.

Reply:
is this what you mean?